jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/LoadDockerStep.java

```diff
@@ -18,91 +18,48 @@
 import com.google.cloud.tools.jib.docker.DockerClient;
 import com.google.cloud.tools.jib.docker.ImageTarball;
+import com.google.cloud.tools.jib.event.EventHandlers;
 import com.google.cloud.tools.jib.image.Image;
-import com.google.common.collect.ImmutableList;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.google.common.util.concurrent.ListeningExecutorService;
 import java.io.IOException;
 import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
 
 /** Adds image layers to a tarball and loads into Docker daemon. */
-class LoadDockerStep implements AsyncStep<BuildResult>, Callable<BuildResult> {
+class LoadDockerStep implements Callable<BuildResult> {
 
-  private final ListeningExecutorService listeningExecutorService;
   private final BuildConfiguration buildConfiguration;
   private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
 
   private final DockerClient dockerClient;
-
-  private final PullAndCacheBaseImageLayersStep pullAndCacheBaseImageLayersStep;
-  private final ImmutableList<BuildAndCacheApplicationLayerStep> buildAndCacheApplicationLayerSteps;
-  private final BuildImageStep buildImageStep;
-
-  private final ListenableFuture<BuildResult> listenableFuture;
+  private final Image builtImage;
 
   LoadDockerStep(
-      ListeningExecutorService listeningExecutorService,
       BuildConfiguration buildConfiguration,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
       DockerClient dockerClient,
-      PullAndCacheBaseImageLayersStep pullAndCacheBaseImageLayersStep,
-      ImmutableList<BuildAndCacheApplicationLayerStep> buildAndCacheApplicationLayerSteps,
-      BuildImageStep buildImageStep) {
-    this.listeningExecutorService = listeningExecutorService;
+      Image builtImage) {
     this.buildConfiguration = buildConfiguration;
     this.progressEventDispatcherFactory = progressEventDispatcherFactory;
     this.dockerClient = dockerClient;
-    this.pullAndCacheBaseImageLayersStep = pullAndCacheBaseImageLayersStep;
-    this.buildAndCacheApplicationLayerSteps = buildAndCacheApplicationLayerSteps;
-    this.buildImageStep = buildImageStep;
-
-    listenableFuture =
-        AsyncDependencies.using(listeningExecutorService)
-            .addStep(pullAndCacheBaseImageLayersStep)
-            .addStep(buildImageStep)
-            .whenAllSucceed(this);
-  }
-
-  @Override
-  public ListenableFuture<BuildResult> getFuture() {
-    return listenableFuture;
+    this.builtImage = builtImage;
   }
 
   @Override
-  public BuildResult call() throws ExecutionException, InterruptedException {
-    return AsyncDependencies.using(listeningExecutorService)
-        .addSteps(NonBlockingSteps.get(pullAndCacheBaseImageLayersStep))
-        .addSteps(buildAndCacheApplicationLayerSteps)
-        .addStep(NonBlockingSteps.get(buildImageStep))
-        .whenAllSucceed(this::afterPushBaseImageLayerFuturesFuture)
-        .get();
```

Comment:
Confirming: `call()` can run only after `pullAndCacheBaseImageLayersStep` and `buildImageStep` is complete. That is, sub-steps to pull base image layers are scheduled and the target `Image` instance is (being) built.

Actually, `pullAndCacheBaseImageLayersStep` is not necessary because `buildImageStep` can run only after all the base image layers are downloaded and cached.

Then, the core logic (`afterPushBaseImageLayerFuturesFuture`) can run only after all the base + application layers are cached and available and the target `Image` is conclusively built and available.